### PR TITLE
New driver and supporting other mods - GVH5184

### PR DESCRIPTION
**Pull request by ElksInNC**
The following changes are being submited.

Readme - updated to show MAC address format for this driver.
blerry.be - added sample MAC entry to show format for this device
blerry_main.be - added handling for new driver
blerry_model_GVH5184.be - different approach than other drivers due to multiple probes and 2-line data packets
Govspoof_devices.be - added the two-packet sequence for the dummy mac - using modulus of the time entry already used in the spoofing driver.  This ensures the packets do not arrive at the same time and simulates that you can get two sequence1 (or2) packets in a row.  It validates the logic that discovery does not happen until bank 1 and bank 2 have arrived at least once.

Testing:
1) Deleted entity from HA.
2) Restarted HA - ensured entity was gone.
3) Removed references to Govee4Probe from MQTT via MQTT Explorer.
4) Restarted BLE32Solo1 device with above modified _.be files loaded in file system.  For testing the blerry.be contained these entries:
```
var user_config = {'D03232363E5D/1': {'alias': 'Govee-4Probe', 'model': 'GVH5184', 'discovery': true},
                                'D03232XXXXXX/1': {'alias': 'Govee-4Probe-Fake', 'model': 'GVH5184', 'discovery': true}}
```
5) Observed the following:
a) MQTT entries for in homeassistant/sensor and in tele/tasmota_blerry
b) MQTT device was created in HA with all expected entities

6) The helper code was pasted into the berry console on the device.  The fake device was created and updates as well.

Currently the driver code has a counter that will force an update after 30 reports that do not have a change to them.  If there is an update the counter is reset to zero.  This can easily be removed if another approach is desired.

### CODE REVIEW

File: blerry/blerry.be [hunk 1]
```diff
@@ -21,6 +21,7 @@
 var user_config = {'A4C138AAAAAA': {'alias': 'trial_govee5075', 'model': 'GVH5075', 'discovery': true},
                    'A4C138BBBBBB': {'alias': 'other_govee5075', 'model': 'GVH5075', 'via_pubs': false},
                    'A4C138XXXXXX': {'alias': 'govee5183meats', 'model': 'GVH5183'},
+                   'D03232XXXXXX/1': {'alias': 'govee5184-4probe-meats', 'model': 'GVH5183'},
```
Comment (tony-fav) [inline blerry/blerry.be:24]: fix to be GVH5184

File: Govee-4Probe-RevEng-Discussion.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+Working on the code for the 4-probe Govee driver.  NOTE - any code is intended as psuedo-code and is not syntax correct.
```
Comment (tony-fav) [inline Govee-4Probe-RevEng-Discussion.md:1]: lets make a folder called `docs` for this and rename it `GVH5184.md`

File: blerry/blerry_model_GVH5184.be [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        output_map['RSSI'] = value['RSSI']
+        if this_device['via_pubs']
+          output_map['Time_via_' + device_topic] = output_map['Time']
+          output_map['RSSI_via_' + device_topic] = output_map['RSSI']
+        end
+        output_map['Battery'] = math.ceil(this_full_data[0]/255.0*100.0)
+        for j:1 .. 2
+          for k:j-1 .. j
+            var probeset
+            if this_full_data[j][3+((k-j)*3)] == 198
```
Comment (tony-fav) [inline blerry/blerry_model_GVH5184.be:61]: `198 = 0xC6 = 0b110001100`
`134 = 0x86 = 0b100001100`

What is the value when not plugged in? It looks like the alarm is the 7th bit which could be detected with
`this_full_data[j][3+((k-j)*3)] & 0x40`

Check [here](https://github.com/tony-fav/tasmota-blerry/blob/6ccd412caad65ed2dcdca485d3dccfce5eb38348/blerry/blerry_model_WoContact.be#L25) where I have done some bit based checks in the Switchbot driver.

Comment (ElksInNC) [reply blerry/blerry_model_GVH5184.be:61]: Cool.
Alarm  		198 = 0xC6 = 0b110001100
Normal 		134 = 0x86 = 0b100001100
Unplugged	008 = 0x08 = 0b000001000

Bit 7 = 1 for Alarm
Bit 3 & 8 = 1 for Probe present

Comment (ElksInNC) [reply blerry/blerry_model_GVH5184.be:61]: Actually

```
Alarm     198     0xC6 = 0b11000110 
Normal    134     0x86 = 0b10000110
Unplug    008     0x08 = 0b00001000
```

Bit 2 - on when probe inserted
Bit 3 - on when probe inserted
Bit 4 - on when probe not present
Bit 7 - on when alarm status
Bit 8 - on when probe inserted

File: docs/GVH5184.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      current_value[sequence_index+4] = .get((15+sequence_index),-2)
+      current_value[sequence_index+5] = .get((17+sequence_index),-2)
+```
+
+7) Now we can compare current_value to last_value.  Both contain a complete set of 12 discrete values.  If the same return 0.
+
+8) If they are not the same we will need to publish the updated data.  At this point - it is unclear if there is any advantage or disadvantage of publishing
+all 12 pieces of data.  If we do discovery on Probe1/2 and Probe3/4 separtely and publish values for each separately - then we would not have entities for 
+Probe 3/4 if they are not plugged in.  TBD.
+they
```
Comment (tony-fav) [inline docs/GVH5184.md:145]: ?

File: blerry/blerry_model_GVH5184.be (hunk 3 above)
Comment (tony-fav) [reply blerry/blerry_model_GVH5184.be:61]: Okay. Obviously what you have is fine, but I'll do some thinking on if treating it with the bits could reduce the code much.

File: blerry/blerry_model_GVH5184.be [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        output_map['mac'] = value['mac']
+        output_map['via_device'] = device_topic
+        output_map['RSSI'] = value['RSSI']
+        if this_device['via_pubs']
+          output_map['Time_via_' + device_topic] = output_map['Time']
+          output_map['RSSI_via_' + device_topic] = output_map['RSSI']
+        end
+        output_map['Battery'] = math.ceil(this_full_data[0]/255.0*100.0)
+        for j:1 .. 2
+          for k:j-1 .. j
```
Comment (tony-fav) [inline blerry/blerry_model_GVH5184.be:59]: This loop structure kind of confuses me.
I probably would have done like a standard looking loop and done the index calculations based on that
```
for j:1..2
  for k:1..2
    2*(j-1)+k would be the probe's number
  end
end
```

File: blerry/blerry_model_GVH5184.be (hunk 3 above)
Comment (tony-fav) [reply blerry/blerry_model_GVH5184.be:61]: What happens when an alarm is triggered but you press the physical button to silence it?

**Pull request merged**